src/EnergyPlus/HeatBalanceAirManager.cc

```diff
@@ -4592,6 +4526,17 @@ void GetRoomAirModelParameters(EnergyPlusData &state, bool &errFlag) // True if
         // this used to be an if (NumOfAirModels == 0) block, but both the IF and the ELSE had the same content, these two lines:
         state.dataRoomAirMod->AirModel(ZoneNum).AirModelName = "MIXING AIR MODEL FOR " + state.dataHeatBal->Zone(ZoneNum).Name;
         state.dataRoomAirMod->AirModel(ZoneNum).ZoneName = state.dataHeatBal->Zone(ZoneNum).Name;
+        // set global flag for non-mixing model
+        if (state.dataRoomAirMod->AirModel(ZoneNum).AirModelType != DataRoomAirModel::RoomAirModel::Mixing) {
+            state.dataRoomAirMod->anyNonMixingRoomAirModel = true;
+        }
+    }
+
+    if (state.dataRoomAirMod->anyNonMixingRoomAirModel) {
+        if (state.dataHeatBal->doSpaceHeatBalanceSimulation || state.dataHeatBal->doSpaceHeatBalanceSizing) {
```

Comment:
There are a few of these and they are all the same. I would expect a little bit of improvement if these were converted to a bool in getInput and used here instead. `if (state.dataHeatBal->doSpaceHeatBalanceSimulationOrSizing) {`. And since I only see doSpaceHeatBalanceSizing in these few conditionals, that bool could be removed (i.e., that bool would no longer be used anywhere if you were to implement this suggestion).